### PR TITLE
Instagram and FB Parser

### PR DESCRIPTION
**Pull request by VamsiMadhur**

### CODE REVIEW

File: userdata_mining/mining/facebook.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
+import os
+import json
+
+
+def parse_fb_advitisors(user, data_path='.'):
```
Comment (yrahul3910) [inline userdata_mining/mining/facebook.py:5]: please rename

File: userdata_mining/mining/facebook.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    apps_and_websites = data['app_posts_v2']
+    for item in apps_and_websites:
+        titles.append(item['title']) 
+    
+    return titles
+
+
+def parse_fb_your_topics(user, data_path='.'):
+    """
+    Mines
```
Comment (yrahul3910) [inline userdata_mining/mining/facebook.py:80]: ??

File: userdata_mining/mining/facebook.py [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+        data = json.load(f)
+
+    # The conversation key only has metadata, skip it.
+    advertisers = data['inferred_topics_v2']
+    
+    return advertisers
+
+def parse_fb_comments(user, data_path='.'):
+    """
+    Mines
```
Comment (yrahul3910) [inline userdata_mining/mining/facebook.py:102]: ??

File: userdata_mining/mining/instagram.py [hunk 4]
```diff
@@ -0,0 +1,743 @@
+import os
+import json
+
+
+def parse_insta_ads_viewed(user, data_path='.'):
+    """
+    Mines
```
Comment (yrahul3910) [inline userdata_mining/mining/instagram.py:7]: ?

**Review by yrahul3910 — CHANGES_REQUESTED** [review]
Please flesh out your comments, and rename.

**Review by yrahul3910 — CHANGES_REQUESTED** [review]
Nice work! In the FbInstaDataMiner class, please return the embeddings as done in the GoogleDataMiner class.

File: userdata_mining/mining/miner.py [hunk 5]
```diff
@@ -60,6 +88,548 @@ def mine_data(self, data_path='.'):
...
+        if fb_messages_data:
+            self.fb_messages_data = [
+                embedding.embed(x) for x in fb_messages_data]
+
+            # Cache embeddings
+            with open(f'{self.data_path}/saved/embeddings/apps.pickle', 'wb') as f:
+                pickle.dump(self.apps_embeddings, f)
+        else:
+            self.apps_embeddings = []
+
```
Comment (nickrgarner) [inline userdata_mining/mining/miner.py:584]: Missing blocks for loading cached embeddings if data is None

Comment (nickrgarner) [inline userdata_mining/mining/miner.py:584]: Missing printout of data details

**Review by nickrgarner — CHANGES_REQUESTED** [review]
Need to import facebook.py and instagram.py into `userdata_mining/mining/__init__.py`. See https://github.com/nickrgarner/google-takeout-data-mining/pull/9/files

**Review by yrahul3910 — CHANGES_REQUESTED** [review]
In userdata_mining/mining/__init__.py, please add a newline at the end of the file, and I will approve.

**Review by nickrgarner — APPROVED** [review]
Good work! Thanks for addressing all the changes

**Review by nickrgarner — APPROVED** [review]

**Review by nickrgarner — APPROVED** [review]

**Review by yrahul3910 — APPROVED** [review]

**Pull request merged**